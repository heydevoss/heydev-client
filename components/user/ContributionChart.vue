<template>
    <vue-no-ssr>
        <div class="chart">
            
            <span class="title"> Contributions </span> 

            <br><chartjs-doughnut
                v-bind:bind="true"
                v-bind:datasets="datasets"
                v-bind:labels="labels"
                v-bind:option="options"
            />
        </div>
    </vue-no-ssr>
</template>

<script>
export default {
    data() {
        return {
            datasets: [
                {
                data: [ this.totalPullRequests, this.totalCommits, this.totalIssues],
                backgroundColor: ["#b388ff", "#2196f3", "#80d8ff"],
                hoverBackgroundColor: ["#673ab7", "#0000FF", "#00FFFF"]
                }
            ],
            labels: ["Pull Requests", "Commits", "Issues"],
            options: {
                legend: {
                    position: 'right'
                }
            }
        }
    },
    props: {
        totalPullRequests: Number,
        totalCommits: Number,
        totalIssues: Number
    }
};
</script>
<style>
.title{
    text-align: center;
    padding: 10px;
    font-size: 25px;
    color: black;
}

.chart{
    display: flex;
    flex-direction: column;
    align-items:baseline; 
    width: 500px;
    height: 500px;

}
</style>

