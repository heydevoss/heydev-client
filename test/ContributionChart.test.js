import { mount } from '@vue/test-utils'
import ContributionChart from '@/components/user/ContributionChart.vue'

describe('ContributionChart', () => {
  const wrapper = mount(ContributionChart)

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders the chartjs-doughnut ', () => {
    expect(wrapper.html()).toContain('<chartjs-doughnut')
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
