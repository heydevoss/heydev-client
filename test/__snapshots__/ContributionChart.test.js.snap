// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributionChart renders correctly 1`] = `
<vue-no-ssr>
  <div>
    <br />
     
    <chartjs-doughnut
      bind="true"
      class="chart"
      datasets="[object Object]"
      labels="Pull Requests Open,Pull Requests Closed,Issues Open,Issues Closed,Commits"
      option="[object Object]"
    />
  </div>
</vue-no-ssr>
`;
